xdsl/dialects/experimental/air.py

```diff
@@ -0,0 +1,810 @@
+        printer.print(")")
+
+        if self.herd_operands:
+            printer.print(" args")
+            printer.print("(")
+            if len(self.sizes) == 1:
+                printer.print("%\\ext0 = ")
+            if len(self.sizes) == 2:
+                printer.print("%ext0 = ")
+                printer.print(self.herd_operands[0])
+                printer.print(", ")
+                printer.print("ext1 = ")
+                printer.print(self.herd_operands[1])
+            if len(self.sizes) == 3:
+                printer.print("%ext0 = ")
+
+        printer.print(" : ")
+        for n_arg in range(len(self.herd_operands)):
+            printer.print(self.herd_operands[n_arg].type)
+            if n_arg < len(self.herd_operands) - 1:
+                printer.print(",")
+
+        printer.print(")")
+        if self.region:
+            printer.print_region(self.region)
+
+    @classmethod
+    def parse(cls, parser: Parser) -> HerdOp:
+        sym_name = parser.parse_optional_symbol_name()
+        async_dependencies: list[Operation | SSAValue] = []
+        if parser.parse_optional_keyword("async"):
+            parser.parse_optional_characters("[")
+            while not parser.parse_optional_characters("]"):
+                async_dependencies.append(parser.parse_operand())
+                parser.parse_optional_characters(",")
+        parser.parse_keyword("tile")
+        arg_list = parser.parse_op_args_list()
+        parser.parse_keyword("in")
+
+        parser.parse_characters("(")
+        tile_size_lst: list[Operation | SSAValue] = []
+        for n_arg in range(len(arg_list)):
+            parser.parse_optional_argument(False)
+            parser.parse_characters("=")
+            tile_size = parser.parse_operand()
+            tile_size_lst.append(tile_size)
+            if n_arg < len(arg_list) - 1:
+                parser.parse_characters(",")
+
+        parser.parse_characters(")")
+        arguments_lst: list[Parser.Argument] = []
+
+        operands_lst: list[Operation | SSAValue] = []
+        if parser.parse_optional_keyword("args"):
+            parser.parse_characters("(")
+            while True:
+                argument = parser.parse_argument(expect_type=False)
+                parser.parse_characters("=")
+                operand = parser.parse_operand()
+
+                # The type of the block argument is not known until the operand has been parsed, e.g. %ext0 = %arg0
+                argument = argument.resolve(operand.type)
+                arguments_lst.append(argument)
+                operands_lst.append(operand)
+
+                if not parser.parse_optional_characters(","):
+                    break
+            parser.parse_characters(")")
+
+            parser.parse_characters(":")
+
+            for n_arg in range(len(operands_lst)):
+                parser.parse_type()
+                if n_arg < len(arg_list) - 1:
+                    parser.parse_optional_characters(",")
```

Comment:
```suggestion
            parser.parse_comma_separated_list(parser.Delimiter.NONE, parser.parse_type)
```
You can add a check on the length if you want, and give a more informative error message, if you care about the length matching